medical/tests/test_medical_patient.py

```diff
@@ -139,3 +139,29 @@ def test_check_birthdate_date(self):
                     now + timedelta(days=20),
                 )
             })
+
+    def test_search_age(self):
+        """
+        When patients are searched by age,
+        it should return patients with the corresponding birth dates
+        """
+        birthdate = datetime.strptime(
+            self.patient_1.birthdate, "%Y-%m-%d"
+        ).date()
+        current_date = date.today()
+        current_year = current_date.year
+        years = current_year - birthdate.year
+        current_day = current_date.day
+        first_possible_birthdate = current_date.replace(
+            year=current_year - (int(years))
+        )
+        last_possible_birthdate = first_possible_birthdate.replace(
+            year=current_year - int(years) + 1,
+            day=current_day - 1
+        )
+        result = self.env['medical.patient'].search(
+            ['&', ('birthdate_date', '>=', first_possible_birthdate),
```

Comment:
You need to search age, which is the column you are adding the search for. You're testing fields were already know work here.